### PR TITLE
:bug: Fix delete control plane bug

### PR DESCRIPTION
**Pull request by chuckha**
I was seeing intermittent delete failures relating to the control plane. This PR aims to improve logging around deleting clusters as well as fixes a bug with the control plane delete logic. This PR will improve the experience for devs who don't know why a cluster isn't deleting by providing them with a list of descendant names.

Signed-off-by: Chuck Ha <chuckh@vmware.com>

Here are some examples of this logging:

```
I0123 18:26:47.891556       1 cluster_controller.go:271] controllers/Cluster "msg"="Cluster still has descendants - need to requeue" "cluster"="test-0" "namespace"="default" "descendants"="Control plane machines: test-0-kvn6l,test-0-wxw6s,test-0-n78fd;Machine deployments: test-0-md;Machine sets: test-0-md-5c6f99b8bf;Worker machines: test-0-md-5c6f99b8bf-hrlvb" "indirect descendants count"=2
I0123 18:26:47.923554       1 cluster_controller.go:271] controllers/Cluster "msg"="Cluster still has descendants - need to requeue" "cluster"="test-0" "namespace"="default" "descendants"="Control plane machines: test-0-kvn6l,test-0-wxw6s,test-0-n78fd;Machine sets: test-0-md-5c6f99b8bf;Worker machines: test-0-md-5c6f99b8bf-hrlvb" "indirect descendants count"=2
I0123 18:26:48.924882       1 cluster_controller.go:271] controllers/Cluster "msg"="Cluster still has descendants - need to requeue" "cluster"="test-0" "namespace"="default" "descendants"="Control plane machines: test-0-kvn6l,test-0-wxw6s,test-0-n78fd;Worker machines: test-0-md-5c6f99b8bf-hrlvb" "indirect descendants count"=1
I0123 18:26:52.908050       1 cluster_controller.go:271] controllers/Cluster "msg"="Cluster still has descendants - need to requeue" "cluster"="test-0" "namespace"="default" "descendants"="Control plane machines: test-0-kvn6l,test-0-wxw6s,test-0-n78fd" "indirect descendants count"=0
```

### CODE REVIEW

File: controllers/cluster_controller.go [hunk 1]
```diff
@@ -314,6 +315,39 @@ func (c *clusterDescendants) length() int {
 		len(c.workerMachines.Items)
 }
 
+func (c *clusterDescendants) descendantNames() string {
```
Comment (chuckha) [inline controllers/cluster_controller.go]: I tried a number of other solutions, but this ended up being the one I liked the most. It's straightforward and did exactly what I wanted it to do. Open to suggestions here if someone feels like tinkering with this sort of refactoring.

File: controllers/cluster_controller.go [hunk 2]
```diff
@@ -209,7 +210,7 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	if cluster.Spec.ControlPlaneRef != nil {
 		obj, err := external.Get(ctx, r.Client, cluster.Spec.ControlPlaneRef, cluster.Namespace)
 		switch {
-		case apierrors.IsNotFound(err):
+		case apierrors.IsNotFound(errors.Cause(err)):
```
Comment (detiber) [inline controllers/cluster_controller.go]: This is a bug that has cropped up on more than one occasion, I'm wondering if we should have a wrapper for this in our errors package (aliased capierrors here) that handles extracting the causing error?

Comment (vincepri) [reply controllers/cluster_controller.go]: Yeah I was thinking the same the other day, we can go the alias route and have one `errors` package that aliases all the things

Comment (chuckha) [reply controllers/cluster_controller.go]: I'm going to open an issue around this. I think we could fix it an alternative way. I don't think this should be a blocker for this PR.

Comment (chuckha) [reply controllers/cluster_controller.go]: https://github.com/kubernetes-sigs/cluster-api/issues/2138

**Pull request merged**